common/utils/storage_utils.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+Package utils contains common helper functions used across multiple other packages.
+Utils should not import any Spanner migration tool packages.
+*/
+package utils
```

Comment:
There is a `dataflow_utils` inside the `accessor` and this one is in `common`. Consider restructuring to have a single, top-level `utils` package.

Reply:
The common/utils package contain methods which do not import any other SMT dependency, only third party, which is how util packages should be. Hence, storage_utils does stuff like parseGCSURI etc. which do not import any other SMT dependency.

The dataflow_utils is a layer that contains methods which call accessors, but are generic enough to be re-used by multiple flows (for ex: unmarshalTuningConfig() will reads local or gcs file, and creates the tuning struct, which will be used by both fwd and reverse so should be in some common package, but not utils since it imports storageaccessor)

The dataflow utils should be renamed/moved to another top level package, a layer of common util-like functions which use accessors, any suggestions?

Reply:
Interesting. I think that makes sense and is fair. 
Probably `utils` is overloaded here? Maybe call the package that houses `unmarshalTuningConfig` as `dataflow_helpers` or something similar. 

I've always viewed `utils` what you said here -

> The common/utils package contain methods which do not import any other SMT dependency, only third party, which is how util packages should be to freely be imported anywhere in SMT. Hence, storage_utils does stuff like parseGCSURI etc. which do not import any other SMT dependency.